### ISSUE TITLE
Pex bootstrap resets PYTHONPATH at runtime when running pex file

### ISSUE THREAD

**Issue by smezouar** [pantsbuild/pex#920]
We use behave in a pex file. Our behave scenario can import external libs defined in PYTHONPATH.
With latest release of pex, the import fails, whereas it was OK with pex 1.6.7

Steps to reproduce:

A .pex file is generated with the following requirements:
```
allure-behave==2.8.11
allure-python-commons==2.8.11
appdirs==1.4.3
attrs==19.3.0
behave==1.2.6
certifi==2019.11.28
distlib==0.3.0
filelock==3.0.12
importlib-metadata==1.5.0 ; python_version < '3.8'
importlib-resources==1.3.1 ; python_version < '3.7'
parse-type==0.5.2
parse==1.15.0
pipenv==2018.11.26
pluggy==0.13.1
six==1.14.0
virtualenv-clone==0.5.3
virtualenv==20.0.10
zipp==3.1.0 ; python_version < '3.8'
```

I have generated the pex file with pex version 1.6.7 and last release 2.1.6:
```
pip install pex==1.6.7
python3.6 -m pex -r requirements.txt -m behave -o behave_pex_1.6.7 --python-shebang /opt/rh/rh-python36/root/usr/bin/python3.6

pip install pex --upgrade
python3.6 -m pex -r requirements.txt -m behave -o behave_pex_2.1.6 --python-shebang /opt/rh/rh-python36/root/usr/bin/python3.6
```

I then install new modules in a custom path with pip
```
pip install -t ./extlibs xmldiff
```
This path is used during the tests done with behave built with pex.
	
Here are the behave feature and step:
```
mkdir features

echo "Feature: Behave pex

  Scenario: Behave PYTHONPATH
    Given I test PYTHONPATH with pex" >  features/mybehave.feature

mkdir steps
echo "from behave import *
import xmldiff

@given(u'I test PYTHONPATH with pex')
def step_impl(context):
  print('OK')
" > steps/mybehave.py
```

Then I run the tests:
```
export PEX_INHERIT_PATH=1
export PYTHONPATH="./extlibs"
./behave_pex_1.6.7
./behave_pex_2.1.6
```

The test with pex 1.6.7 is OK, whereas it fails with pex 2.1.6:
```
Feature: Behave pex # features/mybehave.feature:1

  Scenario: Behave PYTHONPATH        # features/mybehave.feature:3
    Given I test PYTHONPATH with pex # steps/mybehave.py:4 0.001s

1 feature passed, 0 failed, 0 skipped
1 scenario passed, 0 failed, 0 skipped
1 step passed, 0 failed, 0 skipped, 0 undefined
Took 0m0.001s
Exception ModuleNotFoundError: No module named 'xmldiff'
Traceback (most recent call last):
  File "/sources/Framework/pex-behave/behave_pex_2.1.6/.bootstrap/pex/pex.py", line 393, in execute
  File "/sources/Framework/pex-behave/behave_pex_2.1.6/.bootstrap/pex/pex.py", line 325, in _wrap_coverage
  File "/sources/Framework/pex-behave/behave_pex_2.1.6/.bootstrap/pex/pex.py", line 356, in _wrap_profiling
  File "/sources/Framework/pex-behave/behave_pex_2.1.6/.bootstrap/pex/pex.py", line 438, in _execute
  File "/sources/Framework/pex-behave/behave_pex_2.1.6/.bootstrap/pex/pex.py", line 535, in execute_entry
  File "/sources/Framework/pex-behave/behave_pex_2.1.6/.bootstrap/pex/pex.py", line 542, in execute_module
  File "/opt/rh/rh-python36/root/usr/lib64/python3.6/runpy.py", line 208, in run_module
    return _run_code(code, {}, init_globals, run_name, mod_spec)
  File "/opt/rh/rh-python36/root/usr/lib64/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/root/.pex/installed_wheels/0943f08eca1831c061473e7a2ab7182db1f14f74/behave-1.2.6-py2.py3-none-any.whl/behave/__main__.py", line 187, in <module>
    sys.exit(main())
  File "/root/.pex/installed_wheels/0943f08eca1831c061473e7a2ab7182db1f14f74/behave-1.2.6-py2.py3-none-any.whl/behave/__main__.py", line 183, in main
    return run_behave(config)
  File "/root/.pex/installed_wheels/0943f08eca1831c061473e7a2ab7182db1f14f74/behave-1.2.6-py2.py3-none-any.whl/behave/__main__.py", line 127, in run_behave
    failed = runner.run()
  File "/root/.pex/installed_wheels/0943f08eca1831c061473e7a2ab7182db1f14f74/behave-1.2.6-py2.py3-none-any.whl/behave/runner.py", line 804, in run
    return self.run_with_paths()
  File "/root/.pex/installed_wheels/0943f08eca1831c061473e7a2ab7182db1f14f74/behave-1.2.6-py2.py3-none-any.whl/behave/runner.py", line 809, in run_with_paths
    self.load_step_definitions()
  File "/root/.pex/installed_wheels/0943f08eca1831c061473e7a2ab7182db1f14f74/behave-1.2.6-py2.py3-none-any.whl/behave/runner.py", line 796, in load_step_definitions
    load_step_modules(step_paths)
  File "/root/.pex/installed_wheels/0943f08eca1831c061473e7a2ab7182db1f14f74/behave-1.2.6-py2.py3-none-any.whl/behave/runner_util.py", line 412, in load_step_modules
    exec_file(os.path.join(path, name), step_module_globals)
  File "/root/.pex/installed_wheels/0943f08eca1831c061473e7a2ab7182db1f14f74/behave-1.2.6-py2.py3-none-any.whl/behave/runner_util.py", line 386, in exec_file
    exec(code, globals_, locals_)
  File "steps/mybehave.py", line 2, in <module>
    import xmldiff
ModuleNotFoundError: No module named 'xmldiff'
```

**Comment by jsirois** [comment]
The inherit path option changed from a bool to a tristate some time ago. Instead of passing `1`, you'll want `prefer` or `fallback`:
```
$ PEX --help-variables
...
PEX_INHERIT_PATH: String (false|prefer|fallback)

    Allow inheriting packages from site-packages, user site-packages
    and the PYTHONPATH. By default, PEX scrubs any non stdlib packages
    from sys.path prior to invoking the application. Using 'prefer'
    causes PEX to shift any non-stdlib packages before the pex
    environment on sys.path and using 'fallback' shifts them after
    instead. Using this option is generally not advised, but can help
    in situations when certain dependencies do not conform to standard
    packaging practices and thus cannot be bundled into PEX files.
    Default: false.
...
```

I'm going to close this as answered, but please speak up if this doesn't work for you.

**Comment by jsirois** [comment]
Pex could have helped you out here by validating the `PEX_INHERIT_PATH` value. Filed #921 to fix that up.

**Comment by anubhav-nd** [comment]
Hi,

I have been struggling to get `PEX_INHERIT_PATH` working. If I give it any value except `false`, it ends up defaulting to host system python packages. This happens even in the case when I give `fallback` as this variable's value. I have tried giving `--inherit-path=fallback` too as an option while building the pex but that too always defaults to system level modules instead of the one I packaged inside PEX. What am I missing here?
